src/transformers/modeling_utils.py

```diff
@@ -467,11 +467,39 @@ def _end_ptr(tensor: torch.Tensor) -> int:
     return stop
 
 
+def _as_list(value) -> list[str]:
+    if value is None:
+        return []
+    if isinstance(value, str):
+        return [value]
+    if isinstance(value, dict):
+        result: list[str] = []
+        for subvalue in value.values():
+            result.extend(_as_list(subvalue))
+        return result
+    if isinstance(value, Iterable):
+        return list(value)
+    return [value]
+
+
+def _extract_tied_value_names(tied_weight_keys) -> list[str]:
+    if tied_weight_keys is None:
+        return []
+    if isinstance(tied_weight_keys, dict):
```

Comment:
This bit is confusing. AFAICT, `_extract_tied_value_names` is only used in one place, namely below in `_get_tied_weight_keys`:

`value_names = _extract_tied_value_names(list(module._tied_weights_keys.keys()))`

Therefore, the input will always be a list of strings (keys). So why check for `dict` here? Was this supposed to also extract the values?

Moreover, `_as_list` seems to be unnecessary if we only ever pass a list of strings.

Reply:
Will remove it don't worry